dcicutils/misc_utils.py

```diff
@@ -785,6 +801,27 @@ def check_true(test_value, message, error_class=None):
         raise error_class(message)
 
 
+def remove_element(elem, lst, raise_error=True):
+    """
+    Returns a copy of the given list with the first occurrence of the given element removed.
+
+    If the element doesn't occur in the list, an error is raised unless given raise_error=False,
+    in which case a copy of the original list is returned.
+
+    :param elem: an object
+    :param lst: a list
+    :param raise_error: a boolean (default True)
+    """
+
+    result = lst.copy()
+    try:
+        result.remove(elem)
+    except ValueError:
+        if raise_error:
+            raise
```

Comment:
If you're predicating on `raise_error`, shouldn't you catch blanket `Exception` as well?

Reply:
From the doc string: "If the element doesn't occur in the list, an error is raised unless given raise_error=False,
    in which case a copy of the original list is returned." You're saying you think the argument name implies more than the doc string? Some people might agree... I'm not sure if I do.

Reply:
I'm going to leave it as is. If it throws other than `ValueError`, something else is wrong that probably should be flagged. That's the usual reason for being more specific than `Exception`.